.gflows/workflows/build-publish/build-publish.template.yml

```diff
@@ -116,8 +116,30 @@
     #@ end
   #@ end
 
-  #@ if hasattr(data.values,"scan_image"):
-    #@ jobs["scan-image"] = generate_scan_image_job(data.values.scan_image, service_sections)
+  #@ if hasattr(data.values,"services"):
+  #@   for service in getattr(data.values,"services",[]):
+  #@     if hasattr(data.values,"scan_image"):
+  #@       if hasattr(data.values,"main_registry"): 
+  #@         jobs["scan-image-{}".format(service.slug)] = generate_scan_image_job(service.image_name, service_sections, data.values.main_registry)
+  #@       else:
+  #@         if hasattr(data.values,"cache_registry"):
+  #@           jobs["scan-image-{}".format(service.slug)] = generate_scan_image_job(service.image_name, service_sections, data.values.cache_registry)
+  #@         end
+  #@       end
+  #@     end
+  #@   end
+  #@ end
+
+  #@ if hasattr(data.values,"service"):
```

Comment:
we can extract the code to function and reuse it both here and in the loop

Reply:
great point, done :)